admin/components/Home/SiteIdRequested.js

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable jsx-a11y/anchor-is-valid */
+import React from "react";
+import { string, func, shape } from "prop-types";
+import { inject } from "mobx-react";
+import styled from "styled-components";
+import { Box, Heading, Paragraph, FormField, TextInput, Button } from "grommet";
+
+const SiteIdRequested = ({
+  siteId,
+  setSiteId,
+  saveSettings,
+  setSiteIdRequested,
+  titleText,
+  contentText,
+  fieldSiteId,
+  linkText,
+  confirmButtonText,
+  siteIdStatus,
+}) => (
+  <form onSubmit={saveSettings}>
+    <Container margin={{ top: "40px", bottom: "24px" }}>
+      <Header margin={{ horizontal: "0", vertical: "0" }}>{titleText}</Header>
+      <Body>
+        <Comment>{contentText}</Comment>
+        <FormField label={fieldSiteId.label}>
+          <StyledTextInput
+            status={siteIdStatus}
+            placeholder={fieldSiteId.placeholder}
+            value={siteId}
+            onChange={setSiteId}
+          />
+        </FormField>
+      </Body>
+    </Container>
+    <Box direction="row" justify="between" align="center">
+      <Link onClick={setSiteIdRequested}>{linkText}</Link>
+      <Button primary label={confirmButtonText} type="submit" />
+    </Box>
+  </form>
+);
+
+SiteIdRequested.propTypes = {
+  siteId: string.isRequired,
+  setSiteId: func.isRequired,
+  saveSettings: func.isRequired,
+  setSiteIdRequested: func.isRequired,
+  titleText: string.isRequired,
+  contentText: string.isRequired,
+  fieldSiteId: shape({ label: string, placeholder: string }).isRequired,
+  linkText: string.isRequired,
+  confirmButtonText: string.isRequired,
+  siteIdStatus: string,
+};
+
+SiteIdRequested.defaultProps = {
+  siteIdStatus: undefined,
+};
+
+export default inject(({ stores: { settings, languages, ui } }) => {
+  const siteIdRequested = "home.siteIdRequested";
+
+  return {
+    siteId: settings.site_id,
+    setSiteId: event => {
+      ui.setSiteIdStatus();
+      settings.setSiteId(event);
+    },
+    saveSettings: event => {
+      event.preventDefault();
+      ui.validateSettings();
```

Comment:
It looks like settings are being validated and UI shows that validation but they are saved anyway. Is that on purpose?